src/Core/Tool/Authorizer/PostAuthorizer.php

```diff
@@ -140,11 +135,21 @@ public function isAllowed(Entity $entity, $privilege)
             return true;
         }
 
-        // We check if the user is the owner of this post. If so, they are allowed
-        // to do almost anything, **except** change ownership and status of the post, which
-        // only admins can do.
-        if ($this->isUserOwner($entity, $user) && !$entity->hasChanged('user_id')
-            && $privilege !== 'change_status') {
+        // Only admins or users with 'Manage Posts' permission can change status
```

Comment:
I think pulling these first 2 checks out by themselves is a bit clearer, hopefully